### PR TITLE
filter prefixes - option to require all prefixes 

### PR DESCRIPTION
**Pull request by glass-ships**
this PR adds a `require_all` flag to the `filter_prefixes` and `filter_out_prefixes` methods,  
ie. keep/filter out rows which do or don't contain some vs. all of the listed filter_prefixes

### CODE REVIEW

File: sssom/util.py [hunk 1]
```diff
@@ -1170,55 +1170,58 @@ def get_prefixes_used_in_metadata(meta: MetadataType) -> List[str]:
...
+    df: pd.DataFrame,
+    filter_prefixes: List[str],
+    features: list = KEY_FEATURES,
+    require_all_prefixes: bool = True,
 ) -> pd.DataFrame:
-    """Filter any row where a CURIE in one of the key column uses one of the given prefixes.
+    """Filter out rows which do NOT contain a CURIE with a prefix in the filter_prefixes list.
 
-    :param df: Pandas DataFrame
+    Args:
```
Comment (hrshdhgd) [inline sssom/util.py:1206]: `Args:` not needed.

Comment (glass-ships) [reply sssom/util.py:1206]: ah thanks, that was leftover from when i realised this wasn't Google style docstrings!!

**Review by hrshdhgd — CHANGES_REQUESTED** [review]
One minor change, the rest looks good!

**Review by hrshdhgd — APPROVED** [review]

**Comment by glass-ships** [comment]
oh I meant to ask, @hrshdhgd what's the release process for sssompy?

**Comment by hrshdhgd** [comment]
The usual GitHub release process should kick start a PyPI release.

**Comment by glass-ships** [comment]
Oh i see! via github actions? i see docs deployment and built-testing but no publish

**Comment by hrshdhgd** [comment]
So on the home page of the repo, you'll see `Releases` , Click on that and then `Draft New Realease` button. Autogenerate the report and then release. I can do it if needed urgently.

**Comment by glass-ships** [comment]
OH apologies, i misunderstood. Got that going, thanks so much!!

**Comment by hrshdhgd** [comment]
No worries!

**Pull request merged**